ergo/caip19.md

```diff
@@ -0,0 +1,65 @@
+---
+namespace-identifier: ergo-caip19
+title: Ergo Namespace - Assets
+author: Yuriy Gagarin (@gagarin55)
+discussions-to: https://github.com/ChainAgnostic/namespaces/pull/98
+status: Draft
+type: Standard
+created: 2023-11-04
+updated: 2023-11-09
+requires: ["CAIP-2", "CAIP-19"]
+---
+
+## Rationale
+
+Ergo supports custom tokens as first-class citizens as outlined in [EIP-0004].
+
+A transaction can create tokens from thin air in its outputs if the Asset ID matches the ID of the transaction's first input ([box]). Since the box identifier is cryptographically unique, it's impossible to have a second asset with the same identifier. This rule also implies that only one new asset can be created per transaction.
+
+The ID format for fungible (ERC20-equivalent) and non-fungible (ERC-721
+equivalent) tokens is the same.  The process for issuance of either is also the
+same. The only difference is that NFTs must have specific values for certain
+properties.
+
```

Comment:
No, when you see at Ergo Token ID you can't say what kind of token it is. Ethereum has particular ERC for each token standard. But Ergo has only EIP-4 standard which describes additional registers for NFT-tokens. I think this is application task to receive token info from blockchain (or store it somewhere).

Reply:
Cool, there's a few other namespaces like that.  I guess what I am trying to tease out (and suggesting you make explicit in the doc for future implementers with little context) is whether they can append, e.g.  `/1234` for #1234 in a serial NFT series like in `eip155/caip-19` when your application has already retrieved that info from the blockchain before passing that CAIP-19 identifier to another application/counterparty, and conversely when a CAIP-19 received from a trusted counterparty with those additional registers/segments can be trusted versus when they should be validated by checking on-chain context.  not blocking, just suggesting, since CAIP-19 strings are often passed around by folks in contexts quite distant from chains and deducing a lot from the strings themselves, like, for example, in dashboard dapps that aggregate data from various sources while minimizing the required roundtrips to nodes for onchain data.

Reply:
I understand what you are speaking about, but in Ergo there is no well defined NFT standard and particular it is unclear how to implement NFT Collections. There is no Collection/Series ID as we have contract address in Ethereum. 

I suggest to proceed further. If someday the NFT problem in Ergo will have solid solution we are able to update specs.

Reply:
@gagarin55, there is this [EIP-34](https://github.com/ergoplatform/eips/blob/master/eip-0034.md), please check if it is suitable here. If not, please feel free to file a PR with the extension.